src/norm/postgres.nim

```diff
@@ -101,6 +101,20 @@ proc createTables*[T: Model](dbConn; obj: T) =
       fkGroups.add "FOREIGN KEY($#) REFERENCES $#($#)" %
         [obj.col(fld), typeof(get val.model).table, typeof(get val.model).col("id")]
 
+    when obj.dot(fld).hasCustomPragma(fk):
+      # Check val is int
+      when val is not int:
```

Comment:
Shouldn't that be `SomeInteger`? `uint` looks like a valid type for `id` but it's not `int`. It is `SomeInteger` though.

Reply:
I hesitated between the two. But I didn't know if you wanted to allow integer type other than ``int`` to be valid key in norm :).